src/command/stash.spec.ts

```diff
@@ -0,0 +1,150 @@
+import * as temp from 'temp';
+import { createTestRepository, modify, contentIsEqual } from './test-helper';
+import { stash, list, apply, pop, drop, clear } from './stash';
+import { getStatus } from './status';
+import { expect } from 'chai';
+import { WorkingDirectoryFileChange } from '../model/status';
+
+const track = temp.track();
+
+describe('stash', async () => {
+
+    let repositoryPath: string;
+    const testText = 'A modified';
+    const testText2 = 'This is a completely different text.';
+
+    after(async () => {
+        track.cleanupSync();
+    });
+
+    before(async () => {
+        repositoryPath = await createTestRepository(track.mkdirSync());
+    })
+
+    describe('stash', async () => {
+        it('push', async () => {
+            modify(repositoryPath, { path: 'A.txt', data: testText });
+
+            let files: ReadonlyArray<WorkingDirectoryFileChange>;
+            const beforeStatus = await getStatus(repositoryPath);
+            files = beforeStatus.workingDirectory.files;
+            expect(files).to.have.lengthOf(1);
+
+            await stash(repositoryPath);
+
+            const afterStatus = await getStatus(repositoryPath);
+            files = afterStatus.workingDirectory.files;
+
+            expect(files).to.have.lengthOf(0);
+        });
+
+        it('list', async () => {
+            let stashes: string[] = [];
+
+            stashes = await list(repositoryPath);
+
+            expect(stashes).to.have.lengthOf(1);
+        });
```

Comment:
Thank you for the tests! 👍

I do not like that the test cases depend on each other. As I see, you expect to have a stashed item from the `push` suite. Each test cases should be self-contained. In other words, they should pass if we change `before` to `beforeEach` and `after` to `afterEach`.